jplag.frontend-utils/src/main/java/de/jplag/TokenPrinter.java

```diff
@@ -227,14 +213,204 @@ private static List<String> linesFromFile(File file) {
     }
 
     /**
-     * Returns the number of digits (including a minus) of the given number.
+     * This contains all data concerning a line of code in a file and the tokens found in that line.
+     * @param file the file
+     * @param lineNumber the line number inside the file
+     * @param text the code line
+     * @param tokens the tokens found in the code line
+     */
+    private record LineData(String file, Integer lineNumber, String text, List<Token> tokens) {
+
+        /**
+         * Applies the given LineDataConsumer to this LineData.
+         * @param consumer the LineDataConsumer
+         */
+        public void apply(LineDataConsumer consumer) {
+            consumer.apply(file, lineNumber, text, tokens);
+        }
+
+        /**
+         * Convenience method to apply a LineDataConsumer to all lineDatas in the given collection.
+         * @param lineDatas the LineData objects to apply the consumer to
+         * @param consumer the LineDataConsumer to apply
+         */
+        public static void forEachApply(Collection<LineData> lineDatas, LineDataConsumer consumer) {
+            lineDatas.forEach(lineData -> lineData.apply(consumer));
+        }
```

Comment:
Yes, but I thought it looked complicated with two `->` in  the same line.
```java
// TokenPrinter.java::96
lineDatas.forEach(lineData -> lineData.apply((String file, Integer lineNumber, String currentLine, List<Token> tokens) -> {
```
What do you think? Just to reiterate from the documentation, the purpose of the `apply` method is to avoid this:
```java
lineDatas.forEach(lineData -> {
	String file = lineData.file();
	int lineNumber = lineData.lineNumber();
	String currentLine = lineData.currentLine();
	List<Tokens> tokens = lineData.tokens();
	
	// Now, this function really begins.
	// ...
}
```
That's not wrong either, but it distracts from what's really happening.

Reply:
As you access the values each only once (except for the current line), you could even avoid all these variables up-front and just use the accessors where needed. I would prefer that over the current solution as it is just simpler but that is just a subjective opinion. If you prefer your current solution, I am fine with that too.

Reply:
Ah, it doesn't look half as bad as I thought. I'll do just that.